tests/test_gas.py

```diff
@@ -0,0 +1,280 @@
+        with pytest.raises(GASDatasetError):
+            gas_client.get_dataset(dataset_name)
+        gas_client.get_dataset(new_dataset_name)
+
+        gas_client.delete_dataset(new_dataset_name)
+
+    def test_upload_dataset_only_with_file(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        gas_client.create_dataset(dataset_name)
+
+        dataset = Dataset(name=dataset_name)
+        segment = dataset.create_segment("Segment1")
+
+        d = tmp_path / "sub"
+        d.mkdir()
+        for i in range(10):
+            p = d / f"hello{i}.txt"
+            p.write_text("CONTENT")
+            segment.append(Data(local_path=str(p)))
+
+        dataset_client = gas_client.upload_dataset(dataset)
+        assert not dataset_client.get_catalog()
+        segment1 = Segment("Segment1", client=dataset_client)
+        assert len(segment1) == 10
+        assert segment1[0].path == "hello0.txt"
+        assert not segment1[0].label
+
+        gas_client.delete_dataset(dataset_name)
+
+    def test_upload_dataset_with_label(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        gas_client.create_dataset(dataset_name)
+
+        dataset = Dataset(name=dataset_name)
+        segment = dataset.create_segment("Segment1")
+        # When uploading label, upload catalog first.
+        dataset._catalog = Catalog.loads(CATALOG)
+
+        d = tmp_path / "sub"
+        d.mkdir()
+        for i in range(10):
+            p = d / f"hello{i}.txt"
+            p.write_text("CONTENT")
+            data = Data(local_path=str(p))
+            data.label = Label.loads(LABEL)
+            segment.append(data)
+
+        dataset_client = gas_client.upload_dataset(dataset)
+        assert dataset_client.get_catalog()
+        segment1 = Segment("Segment1", client=dataset_client)
+        assert len(segment1) == 10
+        assert segment1[0].path == "hello0.txt"
+        assert segment1[0].label
+
+        gas_client.delete_dataset(dataset_name)
+
+    def test_upload_dataset_to_given_draft(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        dataset_client_1 = gas_client.create_dataset(dataset_name)
+        dataset_client_1.create_draft("test")
+
+        dataset = Dataset(name=dataset_name)
+        segment = dataset.create_segment("Segment1")
+
+        d = tmp_path / "sub"
+        d.mkdir()
+        for i in range(10):
+            p = d / f"hello{i}.txt"
+            p.write_text("CONTENT")
+            segment.append(Data(local_path=str(p)))
+
+        dataset_client_2 = gas_client.upload_dataset(dataset, draft_number=1)
+        segment1 = Segment("Segment1", client=dataset_client_2)
+        assert len(segment1) == 10
+        assert segment1[0].path == "hello0.txt"
```

Comment:
Is it enough to just test the first file ?

Reply:
it's enough😊